Clarify what merge strategy we use to merge the release back into develop (#208)
Babylonpartners/ios-playbook: Cookbook/Technical-Documents/ReleaseProcess.md

   * Make sure you create separate tags (and GitHub releases) for each app released on the AppStore (eg. Babylon 4.1.0 and Telus 4.1.0 would each have their own `babylon/4.1.0` and `telus/4.1.0` tags)
   * Set the body of the GitHub release to the content of the Release Notes for the app
   * Attach the zipped `xcarchive` as an artefact to the GitHub release (if you're using the automated release command, you can find the `*.xcarchive.zip` in the Artifacts top section in the CI build).

> 1. Merge `develop` into the release branch, solve the conflicts (if there are any), and then add the `merge` label — which is set up to `squash and merge`.

Reviewer: ```suggestion
1. Merge `develop` into the release branch, solve the conflicts (if there are any)
1. Open a PR to merge the release branch into `develop`.
1. Add the `merge` label in the PR for the release — which is set up to `squash and merge`.
```

Reviewer: I don't see any value in having `which is set up to `squash and merge``.

Author: I think we can remove the `Open a PR to merge the release branch into `develop` step because it is opened automatically by our bots, if we follow the process

Reviewer: Ok. That sounds good. Let's keep it:
```suggestion
1. Merge `develop` into the release branch, solve the conflicts (if there are any)
1. Add the `merge` label in the PR.
```

Reviewer 2: My take:
```suggestion
1. Merge `develop` into the release branch, and solve the conflicts (if any)
1. Add the `Merge` label to the PR created by the bot to merge the release branch into `develop`.
```

When does the bot create the merge PR btw? Is it triggered by a Slack command? Automatically as soon as we add the tag? Would be nice to mention that in the doc as well.

Reviewer 3: @AliSoftware I believe it's the last step of the TestFlight slack command

Reviewer 4: [nitpick] `solve` -> `resolve` 😸
